libc/string_mem.c

```diff
@@ -0,0 +1,373 @@
+}
+
+
+TEST(string_memmove, no_overlap3)
+{
+	int8_t *ptr = (int8_t *)mem + 60;
+	/* Move so that memory blocks are next to each other*/
+	ret = memmove(ptr, TEST_STR, TEST_STR_SIZE);
+	TEST_ASSERT_EQUAL_PTR(ptr, ret);
+	TEST_ASSERT_EQUAL_MEMORY(TEST_STR, ptr, TEST_STR_SIZE);
+	TEST_ASSERT_EQUAL_MEMORY(TEST_STR, ((int8_t *)mem) + OFFSET, TEST_STR_SIZE);
+	TEST_ASSERT_EACH_EQUAL_INT8(0, ((int8_t *)mem), OFFSET);
+	TEST_ASSERT_EACH_EQUAL_INT8(0, ((int8_t *)mem) + OFFSET + (TEST_STR_SIZE * 2), MEM_SIZE - (OFFSET + (TEST_STR_SIZE * 2)));
+}
+
+
+TEST(string_memmove, whole_overlap)
+{
+	int8_t *ptr = (int8_t *)mem + OFFSET;
+	/* Move at the same address as block is */
+	ret = memmove(ptr, TEST_STR, TEST_STR_SIZE);
+	TEST_ASSERT_EQUAL_PTR(ptr, ret);
+	TEST_ASSERT_EQUAL_MEMORY(TEST_STR, ptr, TEST_STR_SIZE);
+	TEST_ASSERT_EACH_EQUAL_INT8(0, ((int8_t *)mem), OFFSET);
+	TEST_ASSERT_EACH_EQUAL_INT8(0, ((int8_t *)mem) + OFFSET + TEST_STR_SIZE, MEM_SIZE - (OFFSET + TEST_STR_SIZE));
+}
+
+
+TEST(string_memmove, right_overlap)
+{
+	int8_t *ptr = (int8_t *)mem + (OFFSET + OVERLAP_LEN);
+	/* Move data so that data blocks overlap */
+	ret = memmove(ptr, TEST_STR, TEST_STR_SIZE);
+	TEST_ASSERT_EQUAL_PTR(ptr, ret);
+	TEST_ASSERT_EQUAL_MEMORY(TEST_STR, ptr, TEST_STR_SIZE);
+	TEST_ASSERT_EQUAL_MEMORY(TEST_STR, ((int8_t *)mem) + OFFSET, TEST_STR_SIZE - OVERLAP_LEN);
+	TEST_ASSERT_EACH_EQUAL_INT8(0, ((int8_t *)mem), OFFSET);
+	TEST_ASSERT_EACH_EQUAL_INT8(0, ((int8_t *)mem) + OFFSET + TEST_STR_SIZE + OVERLAP_LEN, MEM_SIZE - (OFFSET + TEST_STR_SIZE + OVERLAP_LEN));
+}
+
+
+TEST(string_memmove, left_overlap)
+{
+	int8_t *ptr = (int8_t *)mem + (OFFSET - OVERLAP_LEN);
+	/* Move data so that data blocks overlap */
+	ret = memmove(ptr, TEST_STR, TEST_STR_SIZE);
+	TEST_ASSERT_EQUAL_PTR(ptr, ret);
+	TEST_ASSERT_EQUAL_MEMORY(TEST_STR, ptr, TEST_STR_SIZE);
+	TEST_ASSERT_EQUAL_MEMORY(TEST_STR + OVERLAP_LEN, ptr + TEST_STR_SIZE, OVERLAP_LEN);
+	TEST_ASSERT_EACH_EQUAL_INT8(0, ((int8_t *)mem), OFFSET - OVERLAP_LEN);
+	TEST_ASSERT_EACH_EQUAL_INT8(0, ((int8_t *)mem) + OFFSET + TEST_STR_SIZE, MEM_SIZE - (OFFSET + TEST_STR_SIZE));
+}
+
+
+TEST(string_memmove, move_nothing)
+{
+	int8_t *ptr = (int8_t *)mem + OFFSET;
+	/* Check if any data is moved */
+	ret = memmove(ptr, TEST_STR, 0);
+	TEST_ASSERT_EQUAL_PTR(ptr, ret);
+	TEST_ASSERT_EQUAL_MEMORY(TEST_STR, ptr, TEST_STR_SIZE);
+	TEST_ASSERT_EACH_EQUAL_INT8(0, ((int8_t *)mem), OFFSET);
+	TEST_ASSERT_EACH_EQUAL_INT8(0, ((int8_t *)mem) + OFFSET + TEST_STR_SIZE, MEM_SIZE - (OFFSET + TEST_STR_SIZE));
+}
+
+
+TEST_GROUP(string_memmove_big);
+
+
+TEST_SETUP(string_memmove_big)
+{
+	mem_big = calloc(MEM_SIZE_BIG, 1);
+	TEST_ASSERT_NOT_NULL(mem_big);
+	ret = memcpy(((int8_t *)mem_big) + OFFSET_BIG, big_buf, BIG_SIZE);
+	TEST_ASSERT_EQUAL_PTR(((int8_t *)mem_big) + OFFSET_BIG, ret);
+}
+
+
+TEST_TEAR_DOWN(string_memmove_big)
+{
+	free(mem_big);
+}
+
+
+TEST(string_memmove_big, no_overlap_big1)
```

Comment:
You don't have to suffix test case names by `big` , as the test group is called this way. Please correct those cases like above. 

```suggestion
TEST(string_memmove_big, no_overlap_far)
```